sw/device/tests/entropy_src_fw_override_test.c

```diff
@@ -0,0 +1,407 @@
+  };
+
+  TRY(aes_testutils_setup_encryption(transaction, &aes));
+  AES_TESTUTILS_WAIT_FOR_STATUS(&aes, kDifAesStatusOutputValid, true,
+                                kTimeoutUsec);
+  TRY(aes_testutils_decrypt_ciphertext(transaction, &aes));
+
+  // Run KMAC
+  dif_kmac_key_t software_key = {
+      .share0 = {0x43424140, 0x47464544, 0x4B4A4948, 0x4F4E4D4C, 0x53525150,
+                 0x57565554, 0x5B5A5958, 0x5F5E5D5C},
+      .share1 = {0},
+      .length = kDifKmacKeyLen256,
+  };
+  uint32_t output[8];
+
+  TRY(kmac_testutils_config(&kmac, false));
+  TRY(kmac_testutils_kmac(&kmac, kDifKmacModeKmacLen128, &software_key,
+                          /*custom_string=*/NULL, /*custom_string_len=*/0,
+                          /*message=*/"hello", /*message_len=*/6,
+                          ARRAYSIZE(output), output));
+
+  // Run HMAC.
+  dif_hmac_transaction_t hmac_transaction = {
+      .digest_endianness = kDifHmacEndiannessLittle,
+      .message_endianness = kDifHmacEndiannessLittle,
+  };
+
+  TRY(dif_hmac_mode_sha256_start(&hmac, hmac_transaction));
+  TRY(hmac_testutils_push_message(&hmac, (char *)kHmacRefLongKey,
+                                  sizeof(kHmacRefLongKey)));
+  TRY(hmac_testutils_check_message_length(&hmac, sizeof(kHmacRefLongKey) * 8));
+  TRY(dif_hmac_process(&hmac));
+  dif_hmac_digest_t digest;
+  TRY(hmac_testutils_finish_polled(&hmac, &digest));
+
+  return OK_STATUS();
+}
+
+bool test_main(void) {
+  mmio_region_t base_addr;
+
+  base_addr = mmio_region_from_addr(TOP_EARLGREY_ENTROPY_SRC_BASE_ADDR);
+  CHECK_DIF_OK(dif_entropy_src_init(base_addr, &entropy_src));
+  base_addr = mmio_region_from_addr(TOP_EARLGREY_CSRNG_BASE_ADDR);
+  CHECK_DIF_OK(dif_csrng_init(base_addr, &csrng));
+  base_addr = mmio_region_from_addr(TOP_EARLGREY_EDN0_BASE_ADDR);
+  CHECK_DIF_OK(dif_edn_init(base_addr, &edn0));
+  base_addr = mmio_region_from_addr(TOP_EARLGREY_EDN1_BASE_ADDR);
+  CHECK_DIF_OK(dif_edn_init(base_addr, &edn1));
+  base_addr = mmio_region_from_addr(TOP_EARLGREY_RV_PLIC_BASE_ADDR);
+  CHECK_DIF_OK(dif_rv_plic_init(base_addr, &rv_plic));
+  base_addr = mmio_region_from_addr(TOP_EARLGREY_RV_CORE_IBEX_CFG_BASE_ADDR);
+  CHECK_DIF_OK(dif_rv_core_ibex_init(base_addr, &rv_core_ibex));
+  base_addr = mmio_region_from_addr(TOP_EARLGREY_AES_BASE_ADDR);
+  CHECK_DIF_OK(dif_aes_init(base_addr, &aes));
+  base_addr = mmio_region_from_addr(TOP_EARLGREY_HMAC_BASE_ADDR);
+  CHECK_DIF_OK(dif_hmac_init(base_addr, &hmac));
+  base_addr = mmio_region_from_addr(TOP_EARLGREY_KMAC_BASE_ADDR);
+  CHECK_DIF_OK(dif_kmac_init(base_addr, &kmac));
+
+  LOG_INFO("Configuring interrupts...");
+  configure_interrupts();
+  irq_global_ctrl(true);
+  irq_external_ctrl(false);
+
+  LOG_INFO("Computing EDN parameters");
+  edn_params0 = edn_testutils_auto_params_build(
+      /*disable_rand=*/false, /*res_itval=*/0, /*glen_val=*/0);
+  edn_params1 = edn_testutils_auto_params_build(
+      /*disable_rand=*/false, /*res_itval=*/0, /*glen_val=*/0);
```

Comment:
I think for this test we should use:
```
/*disable_rand=*/true, /*res_itval=*/1, /*glen_val=*/1)
```
This will configure the EDNs/CSRNG instances in a mode where they request maximum entropy from ENTROPY_SRC :-)